tests/parser/test_keyfile.py

```diff
@@ -1894,3 +1894,39 @@ def test_keyfile_ethernet_random_cloned_mac_address(self):
           ethernet.cloned-mac-address: "random"
           ipv4.dns-search: ""
 '''.format(UUID, UUID)})
+
+    def test_veth_pair(self):
+        self.generate_from_keyfile('''[connection]
+id=veth-peer1
+uuid={}
+type=veth
+interface-name=veth-peer1
+
+[veth]
+peer=veth-peer2
+
+[ipv4]
+method=auto\n'''.format(UUID))
+        self.assert_netplan({UUID: '''network:
+  version: 2
+  veths:
+    NM-{}:
+      renderer: NetworkManager
+      dhcp4: true
+      peer: "veth-peer2"
+      networkmanager:
+        uuid: "{}"
+        name: "veth-peer1"
+        passthrough:
+          connection.interface-name: "veth-peer1"
+'''.format(UUID, UUID)})
+
+    def test_veth_without_key_pair(self):
```

Comment:
nitpick (non-blocking): I assume this should be called `veth_peer` instead of `key_pair`?